### PR TITLE
Add support for secret exists

### PR DESCRIPTION
**Pull request by ygalblum**
Add the command along with the abi and tunnel support Add e2e tests
Add man page

#### Does this PR introduce a user-facing change?
Yes

```release-note
New command - podman secret exist <name>
```

### CODE REVIEW

File: pkg/api/server/register_secrets.go
Comment (vrothberg) [inline pkg/api/server/register_secrets.go:106]: I think that a `204` for the positive and a `404` for the negative case would be more RESTful.  That's what the other `/exists` endpoints do as well AFAICS.

**Review by vrothberg — COMMENTED** [review]
Thanks, @ygalblum!

Other than the nit on the http codes, LGTM

File: pkg/api/server/register_secrets.go
Comment (ygalblum) [reply pkg/api/server/register_secrets.go]: Not a samll nit :)
Fixed and adjusted both the server and client sides

**Review by vrothberg — APPROVED** [review]
LGTM

@containers/podman-maintainers PTAL

**Review by Luap99 — COMMENTED** [review]
Can you add a test in `test/apiv2` and actually check that the HTTP code is 204 and 404 just the prevent someone from changing it by accident.

File: cmd/podman/secrets/exists.go
Comment (ygalblum) [reply cmd/podman/secrets/exists.go]: You are correct. Thanks for the catch

**Comment by ygalblum** [comment]
> Can you add a test in `test/apiv2` and actually check that the HTTP code is 204 and 404 just the prevent someone from changing it by accident.

Done

**Review by Luap99 — COMMENTED** [review]
/lgtm
/hold

**Review by vrothberg — COMMENTED** [review]
/lgtm
/hold

**Pull request merged**
